typings on sub-path imports are broken

Basically, this doesn't work:

```js
import { run } from "smoldot/worker"
import { compileBytecode } from "smoldot/bytecode"
```

while this does work:

```js
import { run } from "smoldot/dist/mjs/worker-browser"
import { compileBytecode } from "smoldot/dist/mjs/bytecode-browser"
```

## [DEMO](https://stackblitz.com/edit/stackblitz-starters-9iolhk?file=package.json,src%2Fmain.ts)

> Maybe this is just a typo from you, but you're supposed to import `smoldot/worker` and `smoldot/bytecode`, not `smoldot/worker-browser`/`smoldot/bytecode-browser`.

sorry! I made a typo when creating the issue, but not while writing the code :sweat_smile:

However, I just noticed that the only issue seems to be with the typings. In a normal JS project things seem to work correctly. However, in TS projects I get errors like:

> Cannot find module 'smoldot/bytecode' or its corresponding type declarations

I've created [this POC to demonstrate this issue](https://stackblitz.com/edit/stackblitz-starters-9iolhk?file=package.json,src%2Fmain.ts). In this example we can appreciate how TSC fails, however the js code works correctly (in the `build` script I use `esbuild` to strip the type-definitions, which creates valid js code, which seems to work fine).

So, there is an issue, but just for the TS definitions of the sub-paths.

--- a/wasm-node/javascript/README.md
+++ b/wasm-node/javascript/README.md
@@ -125,3 +125,8 @@ import { compileBytecode } from 'smoldot/bytecode';
 compileBytecode().then((bytecode) => postMessage(bytecode))
 onmessage = (msg) => smoldot.run(msg.data);
 ```
+
+Note that importing sub-paths (for example importing `smoldot/worker`) relies on a relatively
+modern JavaScript feature. If you import a smoldot sub-path from a TypeScript file, you might have
+to configure TypeScript to use `"moduleResolution": "node"`. [The official TypeScript documentation
+itself recommends setting this configuration option](https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies).

--- a/wasm-node/javascript/README.md
+++ b/wasm-node/javascript/README.md
@@ -128,5 +128,7 @@ onmessage = (msg) => smoldot.run(msg.data);
 
 Note that importing sub-paths (for example importing `smoldot/worker`) relies on a relatively
 modern JavaScript feature. If you import a smoldot sub-path from a TypeScript file, you might have
-to configure TypeScript to use `"moduleResolution": "node"`. [The official TypeScript documentation
-itself recommends setting this configuration option](https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies).
+to configure TypeScript to use `"moduleResolution": "node16"`. [The official TypeScript
+documentation itself recommends setting this configuration option to
+`node`](https://www.typescriptlang.org/docs/handbook/module-resolution.html#module-resolution-strategies),
+and it is likely that `node16` becomes the go-to module resolution scheme in the future.

